Feature/eosf 202

sheriefvt opened the pull request:
Ticket:
https://openscience.atlassian.net/browse/EOSF-202

Purpose:
Merge utility function fix-special-char with the current branch.  The utility function will be part of the solution to the EOSF-202 ticket.

sheriefvt committed:
Add fix-special-char util to ember-osf
Ticket ID:EOSF-202

sheriefvt committed:
Fix the test

sheriefvt committed:
fix the test

abought commented:
Thanks; this should be useful!

Two requests before merging:
1. Move the template helper to ember-osf as part of your PR, since it is also of broad general utility, 
2. All new util  functions should have unit tests to confirm they work as expected. For example, you could largely reuse or coopt the unit tests I defined for "deserialize" in my transform: https://github.com/CenterForOpenScience/ember-osf/pull/167/files#diff-593ea87c8af267b3755eeb7a03f81073R27

In general, we strongly encourage all new library code to include YUIDoc docstrings describing what they do and how to use them. This become part of the public human readable documentation:
http://centerforopenscience.github.io/ember-osf/

However, I know you have a lot to do today, and my PR already makes some changes to documentation- so in this case, I will add the documentation after this PR is merged.

abought commented:
I've rebased my branch to account for recent changes, so that your commit history will be a bit cleaner and easier to read. Hope that helps!

abought commented on tests/unit/utils/fix-special-char-test.js:
```diff
@@ -0,0 +1,10 @@
+
+module('Unit | Utility | fix special char');
+
+// Replace this with your real tests.
```
As a general nudge- it's a good idea to remove auto-generated placeholder text like this before you commit a branch.

sheriefvt replied:
Sorry forget to remove..thanks

sheriefvt committed:
Add fixSpecialCharHelper.
Add tests for the new helper.
Update current tests for util function fixSpecialChar

sheriefvt committed:
Adjust indentation

abought commented on tests/unit/helpers/fix-special-char-helper-test.js:
```diff
@@ -0,0 +1,24 @@
+
+    for (let [input, output] of testCases) {
+        let res = fixSpecialCharHelper([input]);
+        assert.equal(res, output);
```
Note to self: Sherif revealed an important edge case bug that I should fix in my own tests after merging: this test failed to catch the fact that in the original version, we passed in `null` and got back `''`.

The solution would be to use this function instead: https://api.qunitjs.com/strictEqual/

abought commented on tests/unit/utils/fix-special-char-test.js:
```diff
@@ -0,0 +1,22 @@
+module('Unit | Utility | fix special char');
+
+const testCases = [
+    ['a regular string', 'a regular string'],
```
Note to self: using the same code over and offer suggests a chance to centralize the logic and avoid duplication. I'll move these fixtures to a shared file after merging.

abought reviewed (approved):
Changes look good, thank you!

After I merge, I will make the changes noted below (including adding documentation). I'll let you know when the final PR is ready for testing.
